pkg/controller/csi_handler_test.go

```diff
@@ -1381,19 +1395,33 @@ func TestCSIHandler(t *testing.T) {
 
 func TestCSIHandlerReconcileVA(t *testing.T) {
 	nID := map[string]string{
-		vaNodeIDAnnotation: testNodeName,
+		vaNodeIDAnnotation: testNodeID,
+	}
+	vaGroupResourceVersion := schema.GroupVersionResource{
+		Group:    storage.GroupName,
+		Version:  "v1",
+		Resource: "volumeattachments",
 	}
-
 	tests := []testCase{
 		{
 			name: "va attached actual state not attached",
 			initialObjects: []runtime.Object{
 				va(true /*attached*/, "" /*finalizer*/, nID /*annotations*/),
 				pvWithFinalizer(),
+				csiNode(),
 			},
 			listerResponse: map[string][]string{
 				// Intentionally empty
 			},
+			expectedActions: []core.Action{
+				// Finalizer is saved first
+				core.NewPatchAction(vaGroupResourceVersion, metav1.NamespaceNone, testPVName+"-"+testNodeName,
```

Comment:
I think that saving of finalizers has been tested thoroughly in `TestCSIHandler` and here we don't need to test it again. So feel free to set it in `initialObjects`.

Reply:
done